.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1895,7 +1895,7 @@ private DateTimeResolutionResult ParseHalfYear(string text, DateObject reference
                 var swift = this.config.GetSwiftYear(orderStr);
                 if (swift < -1)
                 {
-                    return ret;
+                    swift = 0;
```

Comment:
Why is this necessary?

Reply:
otherwise the extraction is parsed only if it contains a year, but this is not always the case e.g. "2º semestre traerá desafíos" (the same behaviour was already implemented in ParseQuarter).